includes/api/shipping-info.php

```diff
@@ -233,6 +233,12 @@ function prepareRatesResponse1cc($package, $vendorId, $orderId, $address)
         $response['shipping_fee']     = isset($response[0]['price']) ? $response[0]['price'] : 0;
         $response['shipping_fee_tax'] = !empty($response[0]['taxes']) ? 0 : 0; //By default tax is considered as zero.
     }
+
+    // check shipping fee for gift card product
+    if(giftCardProduct($orderId)){
+        $response['shipping_fee'] = 0;
+    }
+
```

Comment:
Also, can you please fix this line 
https://github.com/razorpay/razorpay-woocommerce/blob/d97a12ea126829c9e3194069a9e6859553df01cc/includes/api/shipping-info.php#L294
It's adding one more DB operation to all shipping API calls irrespective of whether a merchant using smart cod or not.

Reply:
it is not specific for smart COD , we are using $order->get_total() To verify the min order amount